routes/api/v1/favorites.js

```diff
@@ -0,0 +1,91 @@
+var express = require('express');
+var router = express.Router();
+var fetch = require('node-fetch');
+var User = require('../../../models').User;
+var Location = require('../../../models').Location;
+var Favorite = require('../../../models').Favorite;
+
+/* POST new favorite */
+router.post('/', function(req, res) {
+  var foundUser;
+  var locationName;
+  User.findOne({
+    where: {
+      api_key: req.body.api_key
+    },
+    include: 'locations'
+  })
+  .then(user => {
+    return new Promise((resolve, reject) => {
+      user ? resolve(user) : reject("Invalid API Key")
```

Comment:
this syntax here for the promise looks really clean, i havnt seen that before but it looks smooth.